### PR TITLE
Change lifetime back to a regular input field

### PR DESCRIPTION
**Pull request by huffman**
https://highfidelity.fogbugz.com/f/cases/19995/Create-Lifetime-should-not-be-a-slider

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15343/HighFidelity-Beta-PR14460-0ed448a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15343/HighFidelity-Beta-PR14460-0ed448a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15343/HighFidelity-Beta-Interface-PR14460-0ed448a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15343/HighFidelity-Beta-PR14460-0ed448a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15343/HighFidelity-Beta-Interface-PR14460-0ed448a.exe

File: scripts/system/html/js/entityProperties.js [hunk 1]
```diff
@@ -1182,7 +1182,7 @@ const GROUPS = [
             },
             {
                 label: "Lifetime",
-                type: "number",
+                type: "string",
```
Comment (dback2) [inline scripts/system/html/js/entityProperties.js]: I think we should bring back the code for the old number field, and change the current "number" type to "draggableNumber" type, as opposed to having this as a regular string field.

Comment (huffman) [reply scripts/system/html/js/entityProperties.js]: I was originally going to do that, but I couldn't find a way to disable the up/down button that the number input field has by default. Having the up/down buttons ends up having the same issue as the current draggable number fields.

Comment (dback2) [reply scripts/system/html/js/entityProperties.js]: True didn't think about the up/down steppers. They can be hidden in css with
-webkit-inner-spin-button {
    -webkit-appearance: none;
    visibility: hidden;
}

Comment (huffman) [reply scripts/system/html/js/entityProperties.js]: Have you had success with that? I tried several approaches, I believe including that, each of which had issues.

Comment (dback2) [reply scripts/system/html/js/entityProperties.js]: Yes that is what I used on the input numbers as part of the draggable numbers.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15762/HighFidelity-Beta-PR14460-61673a6.apk

File: scripts/system/html/js/entityProperties.js [hunk 2]
```diff
@@ -1854,7 +1856,33 @@ function createBoolProperty(property, elProperty) {
...
+
+    let elInput = createElementFromHTML(`
+        <input id="${elementID}"
+               class='hide-spinner'
+               type="number"
+               ${propertyData.placeholder ? 'placeholder="' + propertyData.placeholder + '"' : ''}
+               ${propertyData.readOnly ? 'readonly' : ''}></input>
+        `)
+
+
```
Comment (dback2) [inline scripts/system/html/js/entityProperties.js]: If we are bringing number back the same number as before then we should have this here:
```
if (propertyData.min !== undefined) {
	elInput.setAttribute("min", propertyData.min);
}
if (propertyData.max !== undefined) {
	elInput.setAttribute("max", propertyData.max);
}
if (propertyData.step !== undefined) {
	elInput.setAttribute("step", propertyData.step);
}

let defaultValue = propertyData.defaultValue;
if (defaultValue !== undefined) {
	elInput.value = defaultValue;
}
```

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15765/HighFidelity-Beta-PR14460-c10e25d.apk

File: scripts/system/html/js/entityProperties.js [hunk 3]
```diff
@@ -1854,7 +1856,33 @@ function createBoolProperty(property, elProperty) {
...
+    elInput.addEventListener('change', createEmitTextPropertyUpdateFunction(property));
```
Comment (dback2) [inline scripts/system/html/js/entityProperties.js]: We should use the existing createEmitNumberPropertyUpdateFunction which includes multiplier

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15762/HighFidelity-Beta-PR14460-61673a6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15762/HighFidelity-Beta-Interface-PR14460-61673a6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15762/HighFidelity-Beta-PR14460-61673a6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15762/HighFidelity-Beta-Interface-PR14460-61673a6.exe

File: scripts/system/html/js/entityProperties.js [hunk 4]
```diff
@@ -1517,6 +1517,11 @@ function resetProperties() {
             case 'number':
             case 'string': {
                 property.elInput.value = "";
+                if (propertyData.defaultValue !== undefined) {
+                    property.elInput.setValue(propertyData.defaultValue);
```
Comment (dback2) [inline scripts/system/html/js/entityProperties.js]: setValue is a function on draggable numbers, here we would want this instead
```
if (propertyData.defaultValue !== undefined) {
	property.elInput.value = propertyData.defaultValue;
} else { 
	property.elInput.value = "";
}
```

Comment (dback2) [reply scripts/system/html/js/entityProperties.js]: And actually we can remove line 1519 in this case

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15772/HighFidelity-Beta-PR14460-46a258c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15772/HighFidelity-Beta-PR14460-46a258c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15772/HighFidelity-Beta-Interface-PR14460-46a258c.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15779/HighFidelity-Beta-PR14460-25b0425.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15779/HighFidelity-Beta-PR14460-25b0425.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15779/HighFidelity-Beta-Interface-PR14460-25b0425.dmg

**Comment by QliCknox** [comment]
Currently Testing

**Comment by QliCknox** [comment]
Pass

Lifetime is no longer a slider, no property text field accepts non numerical values

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15796/HighFidelity-Beta-PR14460-f42c31c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15796/HighFidelity-Beta-PR14460-f42c31c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15796/HighFidelity-Beta-Interface-PR14460-f42c31c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15796/HighFidelity-Beta-PR14460-f42c31c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15796/HighFidelity-Beta-Interface-PR14460-f42c31c.exe

**Pull request merged**